internal/queue/consumer.go

```diff
@@ -52,7 +53,7 @@ func (q *RabbitQueue) createConsumer(queueData ConsumerQueue) error {
 	return nil
 }
 
-func (q *RabbitQueue) StartConsumer(queueData ConsumerQueue) error {
+func (q *RabbitQueue) StartConsumer(queueData ConsumerQueue, ctx context.Context) error {
```

Comment:
Mostly a nitpick, but can we make ctx the first parameter (in all of the methods where you added it)?